pkg/cmd/server/start/start_master.go

```diff
@@ -451,7 +451,7 @@ func StartAPI(oc *origin.MasterConfig, kc *kubernetes.MasterConfig) error {
 
 	var standaloneAssetConfig *origin.AssetConfig
 	if oc.WebConsoleEnabled() {
-		config, err := origin.BuildAssetConfig(*oc.Options.AssetConfig)
+		config, err := origin.NewAssetConfig(*oc.Options.AssetConfig, oc.Options.KubernetesMasterConfig.AdmissionConfig.PluginConfig)
```

Comment:
How about passing through the exact config you're interested in rather than the entire map?  That would keep the knowledge of how these things are specified segregated.

This is the sort of interaction I was expecting.